downloads.py

```diff
@@ -18,37 +20,59 @@ def is_version_packed(fname):
     return resp.status_code == 200
 
 
-def pack_version(fname, ver_obj, pmgr_obj):
+def pack_version(fname, ver_obj, pmgr_obj, installer_cls):
     if not ver_obj['downloadUrl']:
         return
 
     with zipfile.ZipFile(fname, 'w', zipfile.ZIP_DEFLATED) as ziph:
-        # pack main file
-        download_into_zip(ziph, ver_obj['downloadUrl'], os.path.join("lib", "ext"))
+        tmp_dir = tempfile.mkdtemp()
+
+        # download main jar
+        jar_path = download_into_dir(tmp_dir, ver_obj['downloadUrl'], os.path.join("lib", "ext"))
 
-        # pack libs
+        if installer_cls is not None:
+            os.mkdir(os.path.join(tmp_dir, 'bin'))
+            return_code = subprocess.call(["java", "-cp", jar_path, installer_cls])
```

Comment:
using "check_call" instead of "call" will eliminate the need in following `if`. It's ok to throw exception in case of hard-fail.